packages/engine-server/src/workspace/utils.ts

```diff
@@ -57,4 +68,97 @@ export class WorkspaceUtils {
     }
     return noteHash !== note.contentHash;
   }
+
+  /**
+   * Generate url for given note or return `undefined` if no url is specified
+   * @param opts
+   *
+   */
+  static getNoteUrl(opts: {
+    config: DendronConfig;
+    note: NoteProps;
+    vault: DVault;
+    urlRoot?: string;
+    maybeNote?: NoteProps;
```

Comment:
What's this parameter for?

Reply:
the open document in vscode provided by `CopyNoteUrl` command

Reply:
Ah ok - I see now, this is just a refactor of the code that was inside `CopyNoteUrl`. Instead of having this as a param, in `CopyNoteUrl` why don't we do a check - if maybeNote is undefined, just return and stop the command execution there (and maybe show a warning/msg dialog to the user). Then we can avoid having this parameter in this function - if we're not viewing a note, then there should be no URL to copy.